stackit/internal/services/argus/instance/datasource.go

```diff
@@ -233,23 +233,34 @@ func (d *instanceDataSource) Read(ctx context.Context, req datasource.ReadReques
 		return
 	}
 
-	aclList, err := d.client.ListACL(ctx, instanceId, projectId).Execute()
+	aclListResp, err := d.client.ListACL(ctx, instanceId, projectId).Execute()
 	if err != nil {
 		core.LogAndAddError(ctx, &resp.Diagnostics, "Error reading instance", fmt.Sprintf("Calling API to list ACL data: %v", err))
 		return
 	}
 
+	metricsRetentionResp, err := d.client.GetMetricsStorageRetention(ctx, instanceId, projectId).Execute()
```

Comment:
Why do we need this extra call in the datasource? Don't these fields also come in the `GetInstance` call?

Reply:
Yes, and also the general map fields is already handling this, I think we can remove the mapping entirely

Reply:
Nice catch